redfish-core/lib/pcie_slots.hpp

```diff
@@ -112,7 +114,16 @@ inline void
         slot["HotPluggable"] = *hotPluggable;
     }
 
+    size_t index = slots.size();
```

Comment:
This seems wrong to me. The index is being grabbed before we have copied over the slot array. Doesn't that mean that below we will wipe out one of the slot array elements we move into ["Slots"]?

I would expect index to be set after the move on the next line.

Reply:
Purpose of this line is to remember the last slot `index` so that the content can be added to the same slot in the next logic.
```
    size_t index = slots.size();
    slots.emplace_back(std::move(slot));

    // Get pcie slot location indicator state
    getLocationIndicatorActive(asyncResp, pcieSlotPath,
                               [asyncResp, index](bool asserted) {
        nlohmann::json& slotArray = asyncResp->res.jsonValue["Slots"];
        nlohmann::json& slotItem = slotArray.at(index);    <----
        slotItem["LocationIndicatorActive"] = asserted;
    });
}
```